### DISCUSSION TITLE
Option to reload local storage

### DISCUSSION THREAD

**Discussion by rmalca** [sushinpv/react-secure-storage#33] [Announcements]
Today I ran into a problem that doesn't happen when using localStorage directly.

In the application that I am developing, if there are 2 tabs open and one of them updates the secureLocalStorage, the data is not reflected in the other tab. This is problematic as I use secureLocalStorage to store the access token and if one of the tabs rotates the token the other never finds out unless that tab is refreshed.

I was digging into the code and I think I understand that getAllLocalStorageItems is called only once when loading the page, so I propose that there be a function that allows one and/or all of the localStorage keys to be updated.

**Comment by sushinpv** [comment]
Hi @rmalca , Thank you for your valuable feedback and reporting this issue, 

This particular issue is not very simple to resolve because it requires proper thought process. The main intention which we initially thought to decrypt all the data and keep it on a private variable is for faster data access, so if we want to go with real-time decryption, we need to do a proper retesting of the entire library and a proper load testing to check the performance impact.

This issue has been moved into a discussion where we will prioritize and solve it ASAP. 

Thanks again for your valuable point

**Comment by sushinpv** [comment]
@rmalca We will soon release an alpha version of the library in which you can control the force load / real time decryption using an environment variable, and if the library is stable enough, we will release it to a wider audience.

**Comment by wallace-sf** [comment]
Another important point is that the library should export utility functions. For example, I needed the functions `getDataType` and `getSecurePrefix`. If these functions exist in the library, it would be easier to simply export them. If an issue is created, I am available to collaborate.

**Comment by sushinpv** [comment]
@wallace-sf You are always welcome to contribute!

Note: 

To contribute on the library, make sure you are creating a development branch for your fix as dev/{feature/fix} and create a PR to master branch.

Before creating the PR, Please make sure to remove the react-scripts from the package.json. and you are creating a production build for the library by running yarn build:lib

**Comment by abdullahIsa** [comment]
Hello want to ask if this has been implemented ?

**Comment by dima-vantage** [comment]
Hey there, are there any updates on this?
